src/controllers/hid/hidiothread.cpp

```diff
@@ -0,0 +1,339 @@
+        // Therefore m_outputReportIterator doesn't require Mutex protection.
+        if (m_outputReportIterator->second->sendCachedData(
+                    &m_hidDeviceMutex, m_pHidDevice, m_deviceInfo, m_logOutput)) {
+            // Return after each time consuming sendCachedData
+            return true;
+        }
+    }
+    // Returns false if no report required a time consuming sendCachedData
+    return false;
+}
+
+void HidIoThread::sendFeatureReport(
+        const QByteArray& reportData, unsigned int reportID) {
+    auto startOfHidSendFeatureReport = mixxx::Time::elapsed();
+    QByteArray dataArray;
+    dataArray.reserve(kReportIdSize + reportData.size());
+
+    // Append the Report ID to the beginning of dataArray[] per the API..
+    dataArray.append(reportID);
+
+    for (const int datum : reportData) {
+        dataArray.append(datum);
+    }
+
+    auto lock = lockMutex(&m_hidDeviceMutex);
+    int result = hid_send_feature_report(m_pHidDevice,
+            reinterpret_cast<const unsigned char*>(dataArray.constData()),
+            dataArray.size());
+    if (result == -1) {
+        qCWarning(m_logOutput)
+                << "sendFeatureReport is unable to send data to"
+                << m_deviceInfo.formatName() << "serial #"
+                << m_deviceInfo.serialNumber() << ":"
+                << mixxx::convertWCStringToQString(
+                           hid_error(m_pHidDevice), kMaxHidErrorMessageSize);
+        return;
+    }
+
+    lock.unlock();
+
+    qCDebug(m_logOutput)
+            << result << "bytes sent by sendFeatureReport to"
+            << m_deviceInfo.formatName() << "serial #"
+            << m_deviceInfo.serialNumber() << "(including report ID of"
+            << reportID << ") - Needed: "
+            << (mixxx::Time::elapsed() - startOfHidSendFeatureReport)
+                       .formatMicrosWithUnit();
+}
+
+QByteArray HidIoThread::getFeatureReport(
+        unsigned int reportID) {
+    auto startOfHidGetFeatureReport = mixxx::Time::elapsed();
+    unsigned char dataRead[kReportIdSize + kBufferSize];
+    dataRead[0] = reportID;
+
+    auto lock = lockMutex(&m_hidDeviceMutex);
+    int bytesRead = hid_get_feature_report(m_pHidDevice,
+            dataRead,
+            kReportIdSize + kBufferSize);
+    if (bytesRead <= kReportIdSize) {
+        // -1 is the only error value according to hidapi documentation.
+        // Otherwise minimum possible value is 1, because 1 byte is for the reportID,
+        // the smallest report with data is therefore 2 bytes.
+        qCWarning(m_logInput)
+                << "getFeatureReport is unable to get data from"
+                << m_deviceInfo.formatName() << "serial #"
+                << m_deviceInfo.serialNumber() << ":"
+                << mixxx::convertWCStringToQString(
+                           hid_error(m_pHidDevice), kMaxHidErrorMessageSize);
+        return QByteArray();
```

Comment:
`return {};` works very well in those cases and you can easily spot that a default value is returned. The actual type doesn't matter.